server/proxy/grpcproxy/watch_broadcast.go

```diff
@@ -65,13 +66,46 @@ func newWatchBroadcast(lg *zap.Logger, wp *watchProxy, w *watcher, update func(*
 
 		cctx = withClientAuthToken(cctx, w.wps.stream.Context())
 
-		wch := wp.cw.Watch(cctx, w.wr.key, opts...)
-		wp.lg.Debug("watch", zap.String("key", w.wr.key))
-
-		for wr := range wch {
-			wb.bcast(wr)
-			update(wb)
+		var isRetry bool
+
+	OutLayer:
```

Comment:
The logic is a little bit complicated so I need a few more days to check, sorry for keeping you waiting.